multicluster/cmd/multicluster-controller/clusterproperty_webhook.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2023 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	mcv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+)
+
+//+kubebuilder:webhook:path=/validate-multicluster-crd-antrea-io-v1alpha1-clusterproperty,mutating=false,failurePolicy=fail,sideEffects=None,groups=multicluster.crd.antrea.io,resources=clusterproperties,verbs=create;update;delete,versions=v1alpha1,name=vclusterproperty.kb.io,admissionReviewVersions={v1,v1beta1}
+
+// ClusterProperty validator
+type clusterPropertyValidator struct {
+	Client    client.Client
+	decoder   *admission.Decoder
+	namespace string
+}
+
+// Handle handles admission requests.
+func (v *clusterPropertyValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	clusterProperty := &mcv1alpha1.ClusterProperty{}
+	err := v.decoder.Decode(req, clusterProperty)
+	if err != nil {
+		klog.ErrorS(err, "Error while decoding ClusterProperty", "ClusterProperty", req.Namespace+"/"+req.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	switch req.Operation {
+	case admissionv1.Create:
+		if clusterProperty.Name != mcv1alpha1.WellKnownClusterPropertyClusterSet && clusterProperty.Name != mcv1alpha1.WellKnownClusterPropertyID {
+			return admission.Denied(fmt.Sprintf("name %s is not valid. Only 'cluster.clusterset.k8s.io' and 'clusterset.k8s.io' are valid names for ClusterProperty\n", clusterProperty.Name))
+		}
+	case admissionv1.Update:
+		oldClusterProperty := &mcv1alpha1.ClusterProperty{}
+		if req.OldObject.Raw != nil {
+			if err := json.Unmarshal(req.OldObject.Raw, &oldClusterProperty); err != nil {
+				klog.ErrorS(err, "Error while decoding old ClusterProperty", "ClusterProperty", klog.KObj(clusterProperty))
+				return admission.Errored(http.StatusBadRequest, err)
+			}
+			if oldClusterProperty.Value != clusterProperty.Value {
+				klog.ErrorS(err, "The field 'value' is immutable", "ClusterProperty", klog.KObj(clusterProperty))
+				return admission.Denied("the field 'value' is immutable")
+			}
+			return admission.Allowed("")
+		}
+	case admissionv1.Delete:
+		clusterSetList := &mcv1alpha1.ClusterSetList{}
```

Comment:
This may not be needed, if we remove ClusterSet CRD. Again, we should first make a complete design with the new CRD. @luolanzone